scripts/migrations/2020-10-miro-migration/transfer_packager.py

```diff
@@ -0,0 +1,166 @@
+import concurrent
+import itertools
+import os
+import shutil
+
+from tqdm import tqdm
+
+S3_DOWNLOAD_CONCURRENCY = 3
+
+
+def _assert_local_content_length(file_location, expected_content_length):
+    # Assert we have the correct number of bytes
+    local_content_length = os.path.getsize(file_location)
+
+    # Check that content length is non-zero
+    assert local_content_length > 0
+
+    assert (
+        local_content_length == expected_content_length
+    ), f"Content length mismatch: {file_location}!"
+
+
+def _get_s3_content_length(s3_client, s3_bucket, s3_key):
+    s3_head_object_response = s3_client.head_object(Bucket=s3_bucket, Key=s3_key)
+
+    return s3_head_object_response["ContentLength"]
+
+
+def _download_s3_object(s3_client, s3_bucket, s3_key, target_folder):
+    file_name = s3_key.split("/")[-1]
+    file_location = os.path.join(target_folder, file_name)
+
+    s3_content_length = _get_s3_content_length(
+        s3_client=s3_client, s3_bucket=s3_bucket, s3_key=s3_key
+    )
+
+    def _download():
+        s3_client.download_file(Bucket=s3_bucket, Key=s3_key, Filename=file_location)
+
+    # Check if we already have this file
+    if os.path.isfile(file_location):
+        # If the length doesn't match overwrite
+        if not os.path.getsize(file_location) == s3_content_length:
+            _download()
```

Comment:
I assume this is the bit we want to change to support multiple filenames? Let's discuss exactly how we download files.